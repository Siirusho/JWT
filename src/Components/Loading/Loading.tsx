import { CircularProgress } from '@material-ui/core';
import { FC } from 'react';
import { makeStyles, Theme, createStyles } from "@material-ui/core";

const useStyles = makeStyles<Theme>(theme =>
    createStyles({
        loaderWrapper: {
            position: "absolute",
            transform: "translate(50%, 50%)",
        },
    }));

const Loading: FC = () => {
    const classes = useStyles();
    return (
        <div className={classes.loaderWrapper}>
            <CircularProgress thickness={2} size={400} disableShrink />
        </div>
    );
};

export default Loading;