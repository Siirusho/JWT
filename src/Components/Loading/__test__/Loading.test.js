import React from "react";
import Loading from "../Loading";
import { render } from "@testing-library/react";


describe("Loading component", () => {
    it("should render correct Loading component and match with snapshot", () => {
        const { asFragment } = render(<Loading />);
        expect(asFragment(<Loading />)).toMatchSnapshot();
    })
});