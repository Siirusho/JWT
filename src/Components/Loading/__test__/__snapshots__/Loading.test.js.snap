// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loading component should render correct Loading component and match with snapshot 1`] = `
<DocumentFragment>
  <div
    class="makeStyles-loaderWrapper-1"
  >
    <div
      class="MuiCircularProgress-root MuiCircularProgress-colorPrimary MuiCircularProgress-indeterminate"
      role="progressbar"
      style="width: 400px; height: 400px;"
    >
      <svg
        class="MuiCircularProgress-svg"
        viewBox="22 22 44 44"
      >
        <circle
          class="MuiCircularProgress-circle MuiCircularProgress-circleDisableShrink MuiCircularProgress-circleIndeterminate"
          cx="44"
          cy="44"
          fill="none"
          r="21"
          stroke-width="2"
        />
      </svg>
    </div>
  </div>
</DocumentFragment>
`;
